examples/combobox/js/combobox-autocomplete.js

```diff
@@ -0,0 +1,545 @@
+  if (typeof flag !== 'boolean') {
+    flag = false;
+  }
+
+  if (option) {
+    this.option = option;
+    this.setCurrentOptionStyle(this.option);
+    this.setActiveDescendant(this.option);
+
+    if (this.isBoth) {
+      this.comboboxNode.value = this.option.textContent;
+      if (flag) {
+        this.comboboxNode.setSelectionRange(this.option.textContent.length,this.option.textContent.length);
+      }
+      else {
+        this.comboboxNode.setSelectionRange(this.filter.length,this.option.textContent.length);
+      }
+    }
+  }
+};
+
+ComboboxAutocomplete.prototype.setVisualFocusCombobox = function () {
+  this.listboxNode.classList.remove('focus');
+  this.comboboxNode.parentNode.classList.add('focus'); // set the focus class to the parent for easier styling
+  this.comboboxHasVisualFocus = true;
+  this.listboxHasVisualFocus = false;
+  this.setActiveDescendant(false);
+};
+
+ComboboxAutocomplete.prototype.setVisualFocusListbox = function () {
+  this.comboboxNode.parentNode.classList.remove('focus');
+  this.comboboxHasVisualFocus = false;
+  this.listboxHasVisualFocus = true;
+  this.listboxNode.classList.add('focus');
+  this.setActiveDescendant(this.option);
+};
+
+ComboboxAutocomplete.prototype.removeVisualFocusAll = function () {
+  this.comboboxNode.parentNode.classList.remove('focus');
+  this.comboboxHasVisualFocus = false;
+  this.listboxHasVisualFocus = false;
+  this.listboxNode.classList.remove('focus');
+  this.option = false;
+  this.setActiveDescendant(false);
+};
+
+// ComboboxAutocomplete Events
+
+ComboboxAutocomplete.prototype.filterOptions = function (filter, currentOption) {
+
+  if (typeof filter !== 'string') {
+    filter = '';
+  }
+
+  filter = filter.toLowerCase();
+
+  this.filteredOptions    = [];
+  this.listboxNode.innerHTML = '';
+
+  for (var i = 0; i < this.allOptions.length; i++) {
+    var option = this.allOptions[i];
+    if (filter.length === 0 || option.textComparison.indexOf(filter) === 0) {
+      this.filteredOptions.push(option);
+      var textContent = option.textContent.trim();
+      this.listboxNode.appendChild(option);
+    }
+  }
+
+  // Use populated options array to initialize firstOption and lastOption.
+  var numItems = this.filteredOptions.length;
+  if (numItems > 0) {
+    this.firstOption = this.filteredOptions[0];
+    this.lastOption  = this.filteredOptions[numItems - 1];
+
+    if (currentOption && this.filteredOptions.indexOf(currentOption) >= 0) {
+      option = currentOption;
+    }
+    else {
+      option = this.firstOption;
+    }
+  }
+  else {
+    this.firstOption = false;
+    option = false;
+    this.lastOption  = false;
+  }
+
+  return option;
+};
+
+ComboboxAutocomplete.prototype.setCurrentOptionStyle = function (option) {
+
+  for (var i = 0; i < this.filteredOptions.length; i++) {
+    var opt = this.filteredOptions[i];
+    if (opt === option) {
+      opt.setAttribute('aria-selected', 'true');
+      this.listboxNode.scrollTop = opt.offsetTop;
```

Comment:
I think this was pointed out as being unexpected (point 2 in https://github.com/w3c/aria-practices/issues/1266 ). (That can be fixed separately, just noting it here.)

Reply:
@zcorpan Fixed the option scrolling issue